[Bug] cmake: db/drivers/mysql/dbe.c:69:35: error: ‘MYSQL_OPT_USE_EMBEDDED_CONNECTION’ undeclared

## Describe the bug

The MySQL database drivers uses an undefined option `MYSQL_OPT_USE_EMBEDDED_CONNECTION`.

## To reproduce

1. Install MySQL C connector libraries
2. Configure with `-DWITH_MYSQL=yes`
3. Compile
4. The build fails with

```
/usr/bin/x86_64-pc-linux-gnu-gcc -DGRASS_VERSION_DATE=\"2026\" -DPACKAGE=\"grassmods\" -I/var/tmp/portage/sci-geosciences/grass-8.5.0/work/grass-8.5.0_build/output/lib64/grass85/include -I/var/tmp/portage/sci-geosciences/grass-8.5.0/work/grass-8.5.0/db/drivers/mysql -I/var/tmp/portage/sci-geosciences/grass-8.5.0/work/grass-8.5.0/lib/gis -I/var/tmp/portage/sci-geosciences/grass-8.5.0/work/grass-8.5.0/lib/db/stubs -I/var/tmp/portage/sci-geosciences/grass-8.5.0/work/grass-8.5.0/lib/db/dbmi_base -I/var/tmp/portage/sci-geosciences/grass-8.5.0/work/grass-8.5.0/lib/db/dbmi_driver -isystem /usr/include/mysql  -DGRASS_CMAKE_BUILD=1 -march=native -O2 -pipe -fno-strict-aliasing -std=gnu11 -MD -MT db/drivers/CMakeFiles/mysql.dir/mysql/dbe.c.o -MF db/drivers/CMakeFiles/mysql.dir/mysql/dbe.c.o.d -o db/drivers/CMakeFiles/mysql.dir/mysql/dbe.c.o -c /var/tmp/portage/sci-geosciences/grass-8.5.0/work/grass-8.5.0/db/drivers/mysql/dbe.c
/var/tmp/portage/sci-geosciences/grass-8.5.0/work/grass-8.5.0/db/drivers/mysql/dbe.c: In function ‘db__driver_open_database’:
/var/tmp/portage/sci-geosciences/grass-8.5.0/work/grass-8.5.0/db/drivers/mysql/dbe.c:69:35: error: ‘MYSQL_OPT_USE_EMBEDDED_CONNECTION’ undeclared (first use in this function)
   69 |         mysql_options(connection, MYSQL_OPT_USE_EMBEDDED_CONNECTION, NULL);
      |                                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

## Expected behavior

The build succeeds.

## System description

- Operating System: Gentoo Linux
- GRASS version: 8.5.0
- Build system: CMake
- Compiler: gcc (Gentoo 15.2.1_p20260214 p5) 15.2.1 20260214

## Additional context

In `cmake/find_scripts/FindMySQL.cmake` the build system uses `mysql_config` to discover the MySQL include files and libraries. MySQL doesn't define `MYSQL_OPT_USE_EMBEDDED_CONNECTION`, so the build fails. The option `MYSQL_OPT_USE_EMBEDDED_CONNECTION` is available in the MariaDB client libraries. But these libraries must be discovered by `mariadb_config` instead of `mysql_config`.

I would like to suggest to change `mysql_config` to `mariadb_config` in `cmake/find_scripts/FindMySQL.cmake`, but I am not sure about any further implications.

The bug was discovered through https://bugs.gentoo.org/975254.

--- a/db/drivers/CMakeLists.txt
+++ b/db/drivers/CMakeLists.txt
@@ -18,7 +18,7 @@ set(dbf_SRCS
 set(mysql_SRCS
     mysql/create_table.c
     mysql/cursor.c
-    mysql/dbe.c
+    mysql/db.c
     mysql/describe.c
     mysql/driver.c
     mysql/error.c

